EIPS/eip-5169.md

```diff
@@ -0,0 +1,191 @@
+    event ScriptUpdate(string[] memory newScriptURI);
+
+    /// @notice Get the scriptURI for the contract
+    /// @return The scriptURI
+    function scriptURI() external view returns(string[] memory);
+
+    /// @notice Update the scriptURI 
+    /// emits event ScriptUpdate(scriptURI memory newScriptURI);
+    function setScriptURI(string[] memory newScriptURI) external;
+}
+```
+The interface MUST be implemented under the following constraints:
+
+- The smart contract implementing `IERC5XX0` MUST store variables `address owner` in its state.
+
+- The smart contract implementing `IERC5XX0` MUST set `owner=msg.sender` in its constructor.
+
+- The `ScriptUpdate(...)` event MUST be emitted when the ```setScriptURI``` function updates the `scriptURI`.
+
+- The `setScriptURI(...)` function MUST validate that `owner == msg.sender` *before* executing its logic and updating any state.
+
+- The `setScriptURI(...)` function MUST update its internal state such that `currentScriptURI = newScriptURI`.
+
+- The `scriptURI()` function MUST return the `currentScriptURI` state.
+
+- The `scriptURI()` function MAY be implemented as pure or view.
+
+- Any user of the script learned from `scriptURI` MUST validate the script is either at an immutable location, its URI contains its hash digest, or it implements ERC 5XX1, which asserts authenticity using signatures instead of a digest.
+
+### Rationale
+
+This method avoids the need for building secure and certified centralized hosting and allows scripts to be hosted anywhere: IPFS, GitHub or cloud storage.
+
+### Backwards Compatibility
+
+This standard is compatible with all Token standards (ERC20, 721, 777, 1155 etc.)
+
+### Examples
+
+We here go through a couple of examples of where an authenticated script is relevant for adding additional functionality for tokens. 
+
+1. A Utility NFT is an event ticket and the authenticated script is a JavaScript 'minidapp' which asks the user to sign a challenge message that shows ownership of the key controlling the ticket. The dapp would then render the signature as a QR code which can be scanned by a ticketing app, which could then mark the ticket as used.
+
+2. Smart Token Labs uses a framework called TokenScript; one element of which is a user interface description for contract interaction through tokens.
+Consider a simple 'mint' verb associated with an already existing NFT. The associated script can for example allow the owner to mint a derivative through a contract already holding enough ether for the minting fee, without needing to connect their wallet.
+
+3. An NFT Script which controls a Smartlock. For example consider the lock  being linked to a digital NFT twin and being controlled with the verbs "lock" and "unlock", each of which has an associated JavaScript. Each of these scripts could be executed after the user signs a challenge in a web-view. This is an off-chain example that uses on-chain assets for functionality.
```

Comment:
```suggestion
```

Should either be part of the motivation or be removed.

Reply:
@Pandapip1 removed and reduced into the motivation section as suggested.